pages/docs/observability/sdk/python/instrumentation.mdx

```diff
@@ -326,44 +326,148 @@ with langfuse.start_as_current_span(name="data-processing"):
 
+- the `propagate_attributes` context manager that sets attributes on all observations inside its context and on the trace
 - The `.update_trace()` method on any `LangfuseSpan` or `LangfuseGeneration` object within that trace.
 - `langfuse.update_current_trace()` to update the trace associated with the currently active observation.
 
 **Trace attribute parameters:**
 
-| Parameter    | Type                  | Description                                                      |
-| :----------- | :-------------------- | :--------------------------------------------------------------- |
-| `name`       | `Optional[str]`       | Name for the trace.                                              |
-| `user_id`    | `Optional[str]`       | ID of the user associated with this trace.                       |
-| `session_id` | `Optional[str]`       | Session identifier for grouping related traces.                  |
-| `version`    | `Optional[str]`       | Version of your application/service for this trace.              |
-| `input`      | `Optional[Any]`       | Overall input for the entire trace.                              |
-| `output`     | `Optional[Any]`       | Overall output for the entire trace.                             |
-| `metadata`   | `Optional[Any]`       | Additional metadata for the trace.                               |
-| `tags`       | `Optional[List[str]]` | List of tags to categorize the trace.                            |
-| `public`     | `Optional[bool]`      | Whether the trace should be publicly accessible (if configured). |
+| Parameter    | Type                  | Description                                                      | Recommended Method |
+| :----------- | :-------------------- | :--------------------------------------------------------------- | :----------------- |
+| `name`       | `Optional[str]`       | Name for the trace.                                              | `update_trace()` |
+| `user_id`    | `Optional[str]`       | ID of the user associated with this trace.                       | `propagate_attributes()` |
+| `session_id` | `Optional[str]`       | Session identifier for grouping related traces.                  | `propagate_attributes()` |
+| `version`    | `Optional[str]`       | Version of your application/service for this trace.              | `propagate_attributes()` |
+| `input`      | `Optional[Any]`       | Overall input for the entire trace.                              | `update_trace()` |
+| `output`     | `Optional[Any]`       | Overall output for the entire trace.                             | `update_trace()` |
+| `metadata`   | `Optional[Any]`       | Additional metadata for the trace.                               | `propagate_attributes()` |
+| `tags`       | `Optional[List[str]]` | List of tags to categorize the trace.                            | `update_trace()` |
```

Comment:
We'll also move this to the propagate_attributes(). I think same for `public`. Non-blocking for now, but should follow soonish.